mayhem/scripts/run-mayhem.sh

```diff
@@ -131,27 +131,10 @@ sed -i "s|baseimage:.*|baseimage: $IMAGE_TAG|g" Mayhemfile
 ######################################################################
 # Clear pending runs
 #
-# Fetch the pending or run in progress. These must be stopped for a new run
-# to be started for the same target.
-#
-#  * mayhem show                         << get Mayhem runs
-#  * -n ${MAYHEM_ORGANIZATION}              << limit search to projects in the specified organization
-#  * --format csv                        << display in csv format
-#  * "^openssl/${MAYHEM_TARGET}(?i)/\d+" <<  case-insensitive regex filter by project/target/[run #]
-#  * grep -E "pending|running"           << filter for pending or running runs
-#  * cut -d"," -f1                       << get the run ID from the left-most column in the output
-#  * || true                             << Do not exit the script if no runs match
-######################################################################
-RUNS_TO_STOP=$(${CLI} show -n ${MAYHEM_ORGANIZATION} --format csv "^openssl/${MAYHEM_TARGET}(?i)/\d+" | grep -E "pending|running" | cut -d"," -f1 || true)
-
 # Stop ALL running or pending runs for the selected target. This is required
 # so that the new run is not pending behind previoulsy queued run(s).
-if [[ -n "${RUNS_TO_STOP}" ]]; then
-    for run in $RUNS_TO_STOP
-    do
-        ${CLI} stop -n ${MAYHEM_ORGANIZATION} "${run}" || true
-    done
-fi
+######################################################################
+${CLI} stop -n ${MAYHEM_ORGANIZATION} "openssl/${MAYHEM_TARGET}" || true
```

Comment:
Does this command return when they are stopped, or is stop just a request to stop?

Reply:
The command stops all phases for every matching run prior to returning.

Reply:
Just to probe a bit more: `mayhem run` just requests a run may start; it could fail because of no workers. `mayhem stop` gaurantees?

Reply:
> `mayhem run` just requests a run may start; it could fail because of no workers.

Regardless of the number of workers, the run will get added to the queue. 

> `mayhem stop` gaurantees?

Right - it finds every matching run. For each matching run, the command stops every phase. This applies to runs that are currently running or are queued to run. This is necessary so that we do not enqueue a new run, just to have it stuck behind another run that hasn't started yet.

Reply:
> Regardless of the number of workers, the run will get added to the queue.

Ah, I think this is perhaps the difference in definition.  If the run gets added to a queue todo, would other devops pipeline/tool consider it done?

Reply:
> If the run gets added to a queue todo, would other devops pipeline/tool consider it done?

That depends on whether the run/task is asynchronous or not. If a command is asynchronous (such as `mayhem run`) then most tools will also have a `--wait` argument or something similar to block the caller until the task has completed. 

For instance, the Jenkins CLI has the `-w` option for the `build` command to wait for completion of a job:

```
java -jar jenkins-cli.jar build JOB -w
```

Without the `-w`, it is identical to `mayhem run` -- in that the job is added to the Jenkins queue and the CLI returns immediately.

We _do_ wait for completion of branch builds in `run-mayhem.sh` by chaining `mayhem wait` with `mayhem run` so that the build is blocked until completion.  This is because branch builds are designed to block on regression test failures in this example. The `master` branch does not block, since the the `master` branch is an infinite/very long run.